src/Sarif.Converters.UnitTests/ToolFormatConverterTests.cs

```diff
@@ -109,5 +121,245 @@ public void ToolFormatConverter_TruncatesOutputFileInOverwriteMode()
+            using (var tempDir = new TempDirectory())
+            {
+                const string ToolName = "NonPublicTool";
+                string pluginAssemblyPath = GetCurrentAssemblyPath();
+
+                string inputFilePath = tempDir.Write("input.txt", string.Empty);
+                string outputFilePath = tempDir.Combine("output.txt");
+
+                Action action = () => _converter.ConvertToStandardFormat(
+                    ToolName,
+                    inputFilePath,
+                    outputFilePath,
+                    ToolFormatConversionOptions.None,
+                    pluginAssemblyPath);
+
+                action.ShouldThrow<ArgumentException>()
+                    .Where(ex =>
+                        ex.Message.Contains(pluginAssemblyPath)
+                        && ex.Message.Contains(ToolName));
+            }
+        }
+
+        [TestMethod]
+        public void ToolFileConverter_FailsIfConverterTypeDoesNotHaveCorrectBaseClass()
+        {
+            using (var tempDir = new TempDirectory())
+            {
+                const string ToolName = "IncorrectlyDerivedTool";
+                string pluginAssemblyPath = GetCurrentAssemblyPath();
+
+                string inputFilePath = tempDir.Write("input.txt", string.Empty);
+                string outputFilePath = tempDir.Combine("output.txt");
+
+                Action action = () => _converter.ConvertToStandardFormat(
+                    ToolName,
+                    inputFilePath,
+                    outputFilePath,
+                    ToolFormatConversionOptions.None,
+                    pluginAssemblyPath);
+
+                action.ShouldThrow<ArgumentException>()
+                    .Where(ex =>
+                        ex.Message.Contains(pluginAssemblyPath)
+                        && ex.Message.Contains(ToolName));
+            }
+        }
+
+        [TestMethod]
+        public void ToolFileConverter_FailsIfConverterTypeDoesNotHaveDefaultConstructor()
+        {
+            using (var tempDir = new TempDirectory())
+            {
+                const string ToolName = "NoDefaultConstructorTool";
+                string pluginAssemblyPath = GetCurrentAssemblyPath();
+
+                string inputFilePath = tempDir.Write("input.txt", string.Empty);
+                string outputFilePath = tempDir.Combine("output.txt");
+
+                Action action = () => _converter.ConvertToStandardFormat(
+                    ToolName,
+                    inputFilePath,
+                    outputFilePath,
+                    ToolFormatConversionOptions.None,
+                    pluginAssemblyPath);
+
+                action.ShouldThrow<ArgumentException>()
+                    .Where(ex =>
+                        ex.Message.Contains(pluginAssemblyPath)
+                        && ex.Message.Contains(ToolName));
+            }
+        }
+
+        [TestMethod]
+        public void ToolFormatConverter_FindsConverterInPluginAssembly()
```

Comment:
No. :) So, note: provide tests that provoke IOException, BadImageFormatException, and UnauthorizedAccessException.

---
In reply to: [120773153](https://github.com/Microsoft/sarif-sdk/pull/580#discussion_r120773153) [](ancestors = 120773153)